Full node spec

Manav-Aggarwal opened the pull request:
<!--
Please read and fill out this form before submitting your PR.

Please make sure you have reviewed our contributors guide before submitting your
first PR.
-->

## Overview

<!-- 
Please provide an explanation of the PR, including the appropriate context,
background, goal, and rationale. If there is an issue with this information,
please provide a tl;dr and link the issue. 
-->

## Checklist

<!-- 
Please complete the checklist to ensure that the PR is ready to be reviewed.

IMPORTANT:
PRs should be left in Draft until the below checklist is completed.
-->

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

MSevey commented on node/full_node.md:
```diff
@@ -0,0 +1,40 @@
+
+A Full Node is a top-level service that encapsulates different components of Rollkit and initializes/manages them. The main components are listed in the table below:
+
+ Component     | Description                                                                                                                         |
+ ----------    | ------------------------------------------------------------------------------------------------------------------------------------|
+ proxyApp      | ProxyApp is the interface to the application that consists of multiple connections. It is used to communicate with the application.|
+ genesisDoc    | GenesisDoc is the genesis document that contains information about the initial state of the blockchain.|
+ conf          | Conf is the configuration of the node. It contains all the necessary settings for the node to be initialized and function properly.|
+ P2P           | P2P is the peer-to-peer client used for communication between nodes in the network.|
+ Mempool       | Mempool is the transaction pool where all the transactions are stored before they are added to a block.|
+ Store         | Store is used to store/retrieve blocks, commits, and state.|
+ blockManager  | BlockManager is responsible for managing the operations related to blocks such as creating and validating blocks.|
+ dalc          | DALC is the Data Availability Layer Client used to interact with the data availability layer.|
+ hExService    | HExService is the Header Exchange Service used for exchanging block headers between nodes over P2P.|
+ bExService    | BExService is the Block Exchange Service used for exchanging blocks between nodes over P2P. |
```
We should use the minimum amount of dashes, pipes should be at either end, and use the alignment syntax instead of spaces 

https://www.markdownguide.org/extended-syntax/#tables

```suggestion
| Component | Description |
| :--- | :--- |
| proxyApp | ProxyApp is the interface to the application that consists of multiple connections. It is used to communicate with the application. |
| genesisDoc | GenesisDoc is the genesis document that contains information about the initial state of the blockchain. |
| conf | Conf is the configuration of the node. It contains all the necessary settings for the node to be initialized and function properly. |
| P2P | P2P is the peer-to-peer client used for communication between nodes in the network. |
| Mempool | Mempool is the transaction pool where all the transactions are stored before they are added to a block. |
| Store | Store is used to store/retrieve blocks, commits, and state. |
| blockManager | BlockManager is responsible for managing the operations related to blocks such as creating and validating blocks. |
| dalc | DALC is the Data Availability Layer Client used to interact with the data availability layer. |
| hExService | HExService is the Header Exchange Service used for exchanging block headers between nodes over P2P. |
| bExService | BExService is the Block Exchange Service used for exchanging blocks between nodes over P2P. |
```

MSevey reviewed (changes requested):
Please the corresponding spec symlink.

Also I think we should include information here about what makes the full node different from the light node, i.e. syncing and storing full blocks. 

The table in the Abstract section is good to explain what each component is, I think in the Details section we should explain the how and why more for each component.

Manav-Aggarwal commented:
> Please the corresponding spec symlink.
> 
> Also I think we should include information here about what makes the full node different from the light node, i.e. syncing and storing full blocks.
> 
> The table in the Abstract section is good to explain what each component is, I think in the Details section we should explain the how and why more for each component.

Fixed the symlink-related issue and added a note explaining the difference between a light node and full node.
In order to explain each component more, I found it easier to just rid of the table and just add more info and relevant links in the details to reduce redundancy across the codebase.

Manav-Aggarwal commented:
### Merge Activity

* **Oct 10, 10:33 AM**: @Manav-Aggarwal started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/rollkit/rollkit/1215).
